app/src/main/java/ch/epfl/sweng/SDP/home/HomeActivity.java

```diff
@@ -169,6 +184,40 @@ protected void onCreate(Bundle savedInstanceState) {
         setLeague();
     }
 
+    /**
+     * Checks if storage permissions are granted.
+     * If permissions are revoked it requests permission.
+     * @return a boolean indicating if permissions are granted or not.
+     */
+    public  boolean isStoragePermissionGranted() {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            if (checkSelfPermission(android.Manifest.permission.WRITE_EXTERNAL_STORAGE)
+                    == PackageManager.PERMISSION_GRANTED) {
+                Log.v(TAG,"Permission is granted");
+                return true;
+            } else {
+                Log.v(TAG,"Permission is revoked");
+                ActivityCompat.requestPermissions(this, new String[]{
+                        android.Manifest.permission.WRITE_EXTERNAL_STORAGE}, 1);
+                return false;
+            }
+        } else {
+            //permission is automatically granted on sdk<23 upon installation
+            Log.v(TAG,"Permission is granted");
+            return true;
+        }
+    }
+
+    @Override
+    public void onRequestPermissionsResult(
```

Comment:
You override a method to make a Log? Is this really necessary? (I'm asking because this code is not tested)